Assets/Source/Components/Director/Testing/TestLevel/TestLevelPhase1.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Assets.Source.Components.Base;
+using Assets.Source.Components.Director.Interfaces;
+using Assets.Source.Components.Enemy;
+using Assets.Source.Constants;
+using System.Linq;
+using UnityEngine;
+
+namespace Assets.Source.Components.Director.Testing.TestLevel
+{
+    // todo:  I think we might need a naming scheme to easier tell what order these are in, 
+    // i was thinking numbers but that will make it harder to add things in between phases
```

Comment:
Maybe every ILevelPhase could have a list of ILevelPhasePart or something to make it more granular and each of those phase parts would contains the logic as to what spawns or happens? And then the phaseupdate method would check a condition for the next phase part and once that's ready go to the next one, once all have ran then do flagascomplete. So in a way the levelphase would become a manager of the LevelPhaseParts?

Reply:
Thats not a terrible idea but it makes it hard to have a one phase that branches off into other phases, like if you destroy enough enemies you go to a bonus phase or something, idk.  For now I think its fine how it is, like even if we can't tell the order it'll never get that confusing (theoretically)